pytorch_lightning/plugins/ddp_plugin.py

```diff
@@ -62,3 +63,24 @@ def configure_ddp(self, model, device_ids):
             **self._ddp_kwargs,
         )
         return model
+
+    def init_ddp_connection(
+        self,
+        trainer,
+        cluster_environment,
+        global_rank: int,
+        world_size: int,
+        is_slurm_managing_tasks: bool = True,
```

Comment:
i don't know if this is the right interface. is this leaking too much? should the plugin have a reference to the trainer?

Reply:
IMO we already have too many things with trainer references.  Besides that I feel that this is fine.

Reply:
as mentioned earlier, we’re cleaning up trainer references for the next stage of refactoring. 

but i think that’s for a different PR.